kore/test/Test/Kore/Step/Rule/Simplify.hs

```diff
@@ -301,6 +314,38 @@ test_simplifyRule_OnePathRule =
             )
 
         assertEqual "" expected actual
+
+    , testCase "Case where f(x) in rhs is defined;\
```

Comment:
I added this tests before I started working on the feature, as similar to the lhs case, but now I don't think it's needed anymore because of the updates to https://github.com/kframework/kore/pull/2136/files#diff-b59258b15feafb833a251d4b7ac31319R247